llvm/lib/Target/NVPTX/NVPTXUtilities.cpp

```diff
@@ -179,6 +179,13 @@ static bool argHasNVVMAnnotation(const Value &Val,
   return false;
 }
 
+static std::optional<unsigned> getFnAttrParsedInt(const Function &F,
+                                                  StringRef Attr) {
+  return F.hasFnAttribute(Attr)
+             ? std::optional(F.getFnAttributeAsParsedInteger(Attr))
+             : std::nullopt;
```

Comment:
Ugh.  `{}` can't be used in a ternary, and using std::nullopt forces explicit optional use in the other branch. :-/ 
That ended up being a wash. Sorry about the noise.

Just in case. Comments marked as "nit" are up to you. It includes ignoring them or pushing back when those suggestions don't make sense or turn out not being worth it.

Reply:
No worries! I agree it is basically a wash and will leave it as it currently is.